x/auth/types/account_retriever.go

```diff
@@ -73,7 +74,8 @@ func (ar AccountRetriever) EnsureExists(clientCtx client.Context, addr sdk.AccAd
 func (ar AccountRetriever) GetAccountNumberSequence(clientCtx client.Context, addr sdk.AccAddress) (uint64, uint64, error) {
 	acc, err := ar.GetAccount(clientCtx, addr)
 	if err != nil {
-		return 0, 0, err
+		// if there is an error here it either means its the users first tx. In this case if the user does not have funds the tx will fail anyways
```

Comment:
we should return a typed error and allow the caller to deal with it.

Reply:
the reason i didnt want to change the logic here was to avoid lots of api breaks. I can make the change but dont think its needed. Ill update all docs accordingly though

Reply:
The issue is that GetAccount calls https://github.com/cosmos/cosmos-sdk/blob/139f85133713859d4947bd33ef2a4ac0973344bb/x/auth/types/account_retriever.go#L28, so it may return valid error if the account exists but there is some issues later in the logic.

Reply:
fixed